### PR TITLE
ews workaround

### PR DESCRIPTION
**Pull request by dantavori**
## Contributing to Cortex XSOAR Content
Make sure to register your contribution by filling the [contribution registration form](https://forms.gle/XDfxU4E61ZwEESSMA)

**The Pull Request will be reviewed only after the contribution registration form is filled.**

## Status
- [x] In Progress
- [ ] Ready
- [ ] In Hold - (Reason for hold)

## Related Issues
fixes: https://jira-dc.paloaltonetworks.com/browse/XSUP-35665

## Description
A few sentences describing the overall goals of the pull request's commits.

## Must have
- [ ] Tests
- [ ] Documentation

### CODE REVIEW

**Review by anara123 — APPROVED** [review]

File: Packs/MicrosoftExchangeOnline/Integrations/EWSO365/EWSO365.py [hunk 1]
```diff
@@ -1440,15 +1447,18 @@ def find_folders(client: EWSClient, target_mailbox=None):
     :return: Output tuple
     """
     account = client.get_account(target_mailbox)
-    root = account.root
+    root_collection = FolderCollection(account=account, folders=[account.root])
+    root_folder = account.root
+
     if client.is_public_folder:
-        root = account.public_folders_root
+        root_folder = account.public_folders_root
```
Comment (ShahafBenYakir) [inline Packs/MicrosoftExchangeOnline/Integrations/EWSO365/EWSO365.py:1454]: Wouldn't this fail like here: https://github.com/demisto/content/blob/ea54b979cd8af1c67338535f5654218874482f60/Packs/MicrosoftExchangeOnline/Integrations/EWSO365/EWSO365.py#L347

Comment (JudahSchwartz) [reply Packs/MicrosoftExchangeOnline/Integrations/EWSO365/EWSO365.py:1454]: Dan did this on the next line, so i think that line shouldnt fail. didnt check yet
`            inbox_folders = FolderCollection(account=account, folders=[account.public_folders_root])
`

File: Packs/MicrosoftExchangeOnline/Integrations/EWSO365/EWSO365.py [hunk 2]
```diff
@@ -2500,7 +2505,7 @@ def sub_main():     # pragma: no cover
     params = demisto.params()
     args = prepare_args(demisto.args())
     # client's default_target_mailbox is the authorization source for the instance
-    params['default_target_mailbox'] = args.get('target_mailbox', args.get('source_mailbox', params['default_target_mailbox']))
+    params['default_target_mailbox'] = 'testbox@demistodev.onmicrosoft.com'
```
Comment (JudahSchwartz) [inline Packs/MicrosoftExchangeOnline/Integrations/EWSO365/EWSO365.py:2508]:
```suggestion
    params['default_target_mailbox'] = args.get('target_mailbox', args.get('source_mailbox', params['default_target_mailbox']))
```

Comment (ShahafBenYakir) [inline Packs/MicrosoftExchangeOnline/Integrations/EWSO365/EWSO365.py]:
```suggestion
    args.get('target_mailbox', args.get('source_mailbox', params['default_target_mailbox']))
```

**Comment by daemienb** [comment]
Hey there team! Just wanted to check in and see if there's any news on when we can expect the content pack to be rolled out. It's been causing some issues with our production playbooks. Thanks for your help!

**Review by JudahSchwartz — APPROVED** [review]

**Comment by ShahafBenYakir** [comment]
@daemienb We believe the work is done, once tests will pass we will merge this.

File: Packs/MicrosoftExchangeOnline/ReleaseNotes/1_3_0.md [hunk 3]
```diff
@@ -0,0 +1,6 @@
+
+#### Integrations
+
+##### EWS O365
+
+- Fixed an issue where the integration has failed with 'Not Allowed Access' error due to changes on Microsoft side.
```
Comment (JudahSchwartz) [inline Packs/MicrosoftExchangeOnline/ReleaseNotes/1_3_0.md:6]:
```suggestion
- Fixed an issue where the integration has failed with 'Not Allowed Access' error due to breaking changes on Microsoft's side.
```

**Comment by ShahafBenYakir** [comment]
Validation failure is a false positive - there is no breaking changes on xsoar side

**Pull request merged**